BayesianEstimation/AdaptiveMetropolisWithinGibbs.cs

```diff
@@ -0,0 +1,285 @@
+            SetUpAcceptableParameterRanges();
+            SetUpPriorProbabilityDistributions();
+            SetUpPosteriorFunction(isTwoSample);
+        }
+
+        /// <summary>
+        /// Burn in and then sample the MCMC chain.
+        /// </summary>
+        public void Run(int burnin = 20000, int n = 20000)
+        {
+            n_samples(burnin);
+            n_samples(n);
+        }
+
+        /// <summary>
+        /// Sets up the acceptable range of values for each parameter. For example, standard deviation must be positive.
+        /// </summary>
+        protected void SetUpAcceptableParameterRanges()
+        {
+            if (acceptableParameterRanges == null)
+            {
+                if (isTwoSample)
+                {
+                    acceptableParameterRanges = new List<Tuple<double, double>>[5];
+                    acceptableParameterRanges[0] = new List<Tuple<double, double>> { new Tuple<double, double>(double.NegativeInfinity, double.PositiveInfinity) }; // mu1 (can take any value)
+                    acceptableParameterRanges[1] = new List<Tuple<double, double>> { new Tuple<double, double>(double.NegativeInfinity, double.PositiveInfinity) }; // mu2 (can take any value)
+                    acceptableParameterRanges[2] = new List<Tuple<double, double>> { new Tuple<double, double>(0, double.PositiveInfinity) }; // sd1 (must be positive)
+                    acceptableParameterRanges[3] = new List<Tuple<double, double>> { new Tuple<double, double>(0, double.PositiveInfinity) }; // sd2 (must be positive)
+                    acceptableParameterRanges[4] = new List<Tuple<double, double>> { new Tuple<double, double>(0, double.PositiveInfinity) }; // nu (must be positive)
+                }
+                else
+                {
+                    acceptableParameterRanges = new List<Tuple<double, double>>[3];
+                    acceptableParameterRanges[0] = new List<Tuple<double, double>> { new Tuple<double, double>(double.NegativeInfinity, double.PositiveInfinity) }; // mu (can take any value)
+                    acceptableParameterRanges[1] = new List<Tuple<double, double>> { new Tuple<double, double>(0, double.PositiveInfinity) }; // sd (must be positive)
+                    acceptableParameterRanges[2] = new List<Tuple<double, double>> { new Tuple<double, double>(0, double.PositiveInfinity) }; // nu (must be positive)
+                }
+            }
+        }
+        
+        /// <summary>
+        /// Sets up the prior probability distributions for each parameter.
+        /// </summary>
+        protected void SetUpPriorProbabilityDistributions()
+        {
+            var pooled = Data.SelectMany(p => p.Select(v => v)).ToList();
+
+            meanMu = pooled.Mean();
+            sdMu = pooled.StandardDeviation() * 1000000;
+
+            sigmaLow = pooled.StandardDeviation() / 1000.0;
+            sigmaHigh = pooled.StandardDeviation() * 1000.0;
+
+            exp = 1.0 / 29.0;
```

Comment:
what's this?

Reply:
I changed the location of this, but in this MCMC method (described by Kruschke) the "nu" parameter of the t-distribution has an exponential prior probability distribution. 1/29 is used as the exponent of this prior probability distribution in this example.